content/docs/hooks-state.md

```diff
@@ -212,43 +218,45 @@ Let's now **recap what we learned line by line** and check our understanding.
 
-### Tip: What Do Square Brackets Mean? {#tip-what-do-square-brackets-mean}
+### İpucu: Köşeli Parantez Ne Anlama gelir? {#tip-what-do-square-brackets-mean}
 
-You might have noticed the square brackets when we declare a state variable:
+State değişkenlerini tanımlarken köşeli parantezleri fark etmiş olabilirsiniz:
 
 ```js
   const [count, setCount] = useState(0);
 ```
 
-The names on the left aren't a part of the React API. You can name your own state variables:
+Soldaki isimler React API'ın bir parçası değiller. State değişkeninizi adlandırabilirsiniz:
 
 ```js
   const [fruit, setFruit] = useState('banana');
 ```
 
-This JavaScript syntax is called ["array destructuring"](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Array_destructuring). It means that we're making two new variables `fruit` and `setFruit`, where `fruit` is set to the first value returned by `useState`, and `setFruit` is the second. It is equivalent to this code:
+Bu JavaScript sözdiziminin adı ["array destructuring"](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Array_destructuring). Bunun anlamı `fruit` ve `setFruit` diye iki yeni değişken oluştururken, `fruit`'in yazıldığı yer, ilk değer, `useState` döndürür ve `setFruit`, ikinci, bu koda eşittir: 
+
 
 ```js
   var fruitStateVariable = useState('banana'); // Returns a pair
   var fruit = fruitStateVariable[0]; // First item in a pair
   var setFruit = fruitStateVariable[1]; // Second item in a pair
 ```
 
-When we declare a state variable with `useState`, it returns a pair — an array with two items. The first item is the current value, and the second is a function that lets us update it. Using `[0]` and `[1]` to access them is a bit confusing because they have a specific meaning. This is why we use array destructuring instead.
+`useState` ile state değişkeni tanımladığımız zaman, bu bir çift (iki elemanlı bir dizi) döndürür. İlk eleman o anki değer, ikincisi ise onu güncelleyen fonksiyondur. `[0]` ve `[1]` kullanarak erişmek biraz kafa karıştırıcı çünkü onların kendine özgü anlamları var. Bu yüzden dizi yıkıcıların yerine kullanıyoruz.
 
->Note
+>Not
 >
->You might be curious how React knows which component `useState` corresponds to since we're not passing anything like `this` back to React. We'll answer [this question](/docs/hooks-faq.html#how-does-react-associate-hook-calls-with-components) and many others in the FAQ section.
+>React'ın `useState` bileşenlerinin hangi bileşene karşılık geldiğini nasıl anladığını merak ediyor olabilirsiniz. çünkü `this` gibi bir şeyi React'a göndermedik. Bunu burada cevaplıyoruz: [this question](/docs/hooks-faq.html#how-does-react-associate-hook-calls-with-components) ve birçok şeyi FAQ bölümünde cevaplıyoruz.
```

Comment:
I believe this is more accurate and correct:

`this` gibi bir sey iletmedigimiz icin, React'in `useState`'in hangi bilesene denk geldigini nasil bildigini merak ediyor olabilirsiniz. [Bu soruyu](/docs/hooks-faq.html#how-does-react-associate-hook-calls-with-components) ve bunun gibi daha bir cogunu S.S.S. kisminda cevapliyoruz.